routes/api/v1/favorites.js

```diff
@@ -1,66 +1,82 @@
 const configuration = require('../../../knexfile')[environment];
 const database = require('knex')(configuration);
-const fetch = require('node-fetch');
-const { URL, URLSearchParams } = require('url');
+
 
 router.post('/', async (request, response) => {
-  if (request.body.title && request.body.artistName) {
-    let trackTitle = request.body.title;
-    let artistName = request.body.artistName;
+  let trackTitle = request.body.title;
+  let artistName = request.body.artistName;
 
-    let url = new URL('https://api.musixmatch.com/ws/1.1/track.search');
-    let params = {
-      apikey: process.env.MUSIXMATCH_API_KEY,
-      q_artist: artistName,
-      q_track: trackTitle,
-      s_artist_rating: 'desc'
-    };
-    url.search = new URLSearchParams(params).toString();
+  if (trackTitle && artistName) {
 
     try {
-      let favoriteTrack = await fetch(url);
-      let trackData = await favoriteTrack.json();
+      let trackData = await musixmatchService.getTrackData(trackTitle, artistName, response);
       let track = new Track(trackData);
-
-      let insertToDB = database('favorites').insert(
-        { title: track.title,
-          artistName: track.artistName,
-          genre: track.genre,
-          rating: track.rating
-        }, ['id', 'title', 'artistName', 'genre', 'rating'])
-        .then(track => response.status(200).json(track[0]));
+      let exists = await alreadyFavorite(track, response)
+      if (exists) { return errorResponse(409, exists, response) }
+      await addFavoriteToDB(track, response);
     } catch(error) {
-      console.log(error);
-      response.status(200).json({error: 'There was an error.'});
+      return errorResponse(500, 'Something went wrong. Please try again.', response)
     }
 
-  } else if (request.body.title && !request.body.artistName) {
-    response.status(400).json({error: 'Bad Request! Did you send an artist name?'});
-  } else if (!request.body.title && request.body.artistName) {
-    response.status(400).json({error: 'Bad Request! Did you send a song title?'});
+  } else if (trackTitle && !artistName) {
+    return errorResponse(400, 'Bad Request! Did you send an artist name?', response)
+  } else if (!trackTitle && artistName) {
+    return errorResponse(400, 'Bad Request! Did you send a song title?', response)
   } else {
-    response.status(400).json({error: 'Bad Request! Did you send an artist name and song title?'});
+    return errorResponse(400, 'Bad Request! Did you send an artist name and song title?', response)
   }
 
 });
 
 router.get('/', async (request, response) => {
-  database('favorites')
+  return await database('favorites')
       .then((favorites) => {
         if (favorites.length) {
-          response.status(200).json(favorites);
+          return response.status(200).json(favorites);
         } else {
-          response.status(200).json({ message: 'No favorites found!' });
+          return response.status(200).json({ message: 'No favorites found!' });
```

Comment:
We _might_ get a merge conflict here. So we will have to update this line of code with the updated empty array response.